### DISCUSSION TITLE
"peer" class not working within @apply directive.

### DISCUSSION THREAD

**Discussion by esheltonusef** [tailwindlabs/tailwindcss#14113] [Help] [answered]
I use tailwindcss-windows-x64.exe as stand-alone in C# MVC .net environment.

The build fails if I use this:

`.form-toggle {
    @apply peer;
}`

Whereas this works:

`.form-toggle {
    @apply w-2;
}`

Am I missing something, or does peer not work in this case?

Thanks

**Comment by wongjn** [comment]
`peer` does not work in this case. `peer` is not a class that Tailwind would "know" about. It is used by the `peer-*` variants, but it is not a utility class in and of itself.

---

As an aside, you may also wish to avoid `@apply` altogether as recommended by Adam Wathan, creator of Tailwind:
 - https://twitter.com/adamwathan/status/1226511611592085504
   > Confession: The `apply` feature in Tailwind basically only exists to trick people who are put off by long lists of classes into trying the framework.
   >
   > You should almost never use it 😬
   > 
   > Reuse your utility-littered HTML instead.

 - https://twitter.com/adamwathan/status/1559250403547652097
   > I can say with absolute certainty that if I started Tailwind CSS over from scratch, there would be no `@​apply` 😬
   >
   > The behavior is outrageously complicated, everyone struggles to build the right mental model of what it's supposed to do, and it encourages bad CSS architecture.

**Reply by esheltonusef** [reply]
Thanks for the response, makes sense.

**Comment by adamwathan** [comment, marked as answer]
Like @wongjn mentioned, `peer` isn't actually a utility class, you'll see that using it on it's own generates no CSS:

<img width="764" alt="image" src="https://github.com/user-attachments/assets/7b21ca91-9363-4bb0-a351-e554856dbfdf">

You're probably trying to do something like this right:

```css
.class-1 {
  @apply peer;
}

.class-2 {
  @apply peer-hover:bg-black;
}
```

...and you're expecting that hovering over the `class-1` element would change the background of the `class-2` element.

But this actually would never work the way you'd expect, because look at the CSS generated for `.class-2`:

```css
.peer:hover ~ .class-2 {
  --tw-bg-opacity: 1;
  background-color: rgb(0 0 0 / var(--tw-bg-opacity));
}
```

It includes the `peer` class, not `class-1`, which makes sense when you think about it because that's what `peer-hover:bg-black` does, it looks for `peer`.

So in your case you likely just want to write the actual CSS you need:

```css
.class-1:hover ~ .class-2 {
  @apply bg-black;
}
```

Hope that helps!

**Reply by esheltonusef** [reply]
Thanks for the response, makes perfect sense.
